CRM/Upgrade/Incremental/php/FiveSeventyThree.php

```diff
@@ -29,6 +29,24 @@ class CRM_Upgrade_Incremental_php_FiveSeventyThree extends CRM_Upgrade_Increment
    */
   public function upgrade_5_73_alpha1($rev): void {
     $this->addTask(ts('Upgrade DB to %1: SQL', [1 => $rev]), 'runSql', $rev);
+    $this->addTask(ts('Disable financial ACL extension if unused'), 'disableFinancialAcl');
+  }
+
+  public static function disableFinancialAcl($rev): bool {
+    $setting = CRM_Core_DAO::singleValueQuery('SELECT value FROM civicrm_setting WHERE name = "acl_financial_type"');
+    if (!$setting) {
+      return TRUE;
+    }
+    $setting = unserialize($setting);
+    if ($setting) {
```

Comment:
@eileenmcnaughton I'm confused about what this code is doing. Why the `unserialize()`? The setting appears to be a boolean, not serialized:

https://github.com/civicrm/civicrm-core/blob/aac2d7085ebbb68dac90eca05e04b2188876bc28/ext/financialacls/tests/phpunit/Civi/Financialacls/FinancialTypeTest.php#L24

Reply:
Oh, ok I guess you have to unserialize because you're pulling it directly from the database?
But now I'm still confused because the logic appears to be backward. You are uninstalling the extension if the setting is TRUE? Shouldn't it be the other way 'round?

Reply:
@colemanw yep.... it should be...

Reply:
@colemanw above is fixed